src/kbmod/results.py

```diff
@@ -0,0 +1,591 @@
+    ]
+
+    def __init__(self, data=None, track_filtered=False):
+        """Create a ResultTable class.
+
+        Parameters
+        ----------
+        data : `dict`, `astropy.table.Table`
+        track_filtered : `bool`
+            Whether to track (save) the filtered trajectories. This will use
+            more memory and is recommended only for analysis.
+        """
+        # Set up information to track which row is filtered at which round.
+        self.track_filtered = track_filtered
+        self.filtered = {}
+
+        if data is None:
+            # Set up the basic table meta data.
+            self.table = Table(
+                names=[col[0] for col in self._required_cols],
+                dtype=[col[1] for col in self._required_cols],
+            )
+        elif type(data) is dict:
+            self.table = Table(data)
+        elif type(data) is Table:
+            self.table = data.copy()
+        else:
+            raise TypeError(f"Incompatible data type {type(data)}")
+
+        # Check that we have the correct columns.
+        for col in self._required_cols:
+            if col[0] not in self.table.colnames:
+                raise KeyError(f"Column {col[0]} missing from input data.")
+
+    def __len__(self):
+        return len(self.table)
+
+    def __str__(self):
+        return str(self.table)
+
+    def __repr__(self):
+        return repr(self.table).replace("Table", "Results")
+
+    def _repr_html_(self):
+        return self.table._repr_html_().replace("Table", "Results")
+
+    def __getitem__(self, key):
+        return self.table[key]
+
+    @property
+    def colnames(self):
+        return self.table.colnames
+
+    @classmethod
+    def from_trajectories(cls, trajectories, track_filtered=False):
+        """Extract data from a list of Trajectory objects.
+
+        Parameters
+        ----------
+        trajectories : `list[Trajectory]`
+            A list of trajectories to include in these results.
+        track_filtered : `bool`
+            Indicates whether to track future filtered points.
+        """
+        # Create dictionaries for the required columns.
+        input_d = {}
+        invalid_d = {}
+        for col in cls._required_cols:
+            input_d[col[0]] = []
+            invalid_d[col[0]] = []
+        num_valid = 0
+        num_invalid = 0
+
+        # Add the valid trajectories to the table. If we are tracking filtered
+        # data, add invalid trajectories to the invalid_d dictionary.
+        for trj in trajectories:
+            if trj.valid:
+                input_d["x"].append(trj.x)
+                input_d["y"].append(trj.y)
+                input_d["vx"].append(trj.vx)
+                input_d["vy"].append(trj.vy)
+                input_d["likelihood"].append(trj.lh)
+                input_d["flux"].append(trj.flux)
+                input_d["obs_count"].append(trj.obs_count)
+                num_valid += 1
+            elif track_filtered:
```

Comment:
It is add the valid trajectories to `input_d` whenever it encounters one. It is only adding the invalid trajectories to `invalid_d` if it is going to use track the items it filtered out.

If `track_filtered` is `True`: Then `input_d` is all the valid trajectories and `invalid_d` is all the invalid ones.
Otherwise (when `track_filtered==False`): Then `input_d` is all the valid trajectories and `invalid_d` is empty.

Reply:
Ah, I see. ok, no problems.